src/cloudWatchLogs/document/logStreamDocumentProvider.ts

```diff
@@ -35,6 +40,54 @@ export class LogStreamDocumentProvider implements vscode.TextDocumentContentProv
         }
         return content ?? ''
     }
+
+    async provideDefinition(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        token: vscode.CancellationToken
+    ): Promise<vscode.Definition | vscode.LocationLink[] | undefined> {
+        console.log('provideDefinition')
+        const activeUri = document.uri
+        const logGroupInfo = parseCloudWatchLogsUri(activeUri).logGroupInfo
+        if (logGroupInfo.streamName) {
+            // This means we have a stream file not a log search.
+            return
+        }
+        const curLine = document.lineAt(position.line)
+        let streamUri: vscode.Uri
+        try {
+            const streamIDMap = this.registry.getStreamIdMap(activeUri)
+            if (!streamIDMap || streamIDMap.size === 0) {
+                throw new Error(`cwl: No streamIDMap found for stream with uri ${activeUri}`)
+            }
+            const streamID = streamIDMap.get(curLine.lineNumber)
+
+            if (!streamID) {
+                throw new Error(
+                    `cwl: current line number ${curLine.lineNumber} is unregistered in streamIDMap for ${activeUri}`
+                )
+            }
+            const parameters: CloudWatchLogsParameters = {
+                limit: this.registry.configuration.get('limit', 10000),
+            }
+            logGroupInfo.streamName = streamID
+            streamUri = createURIFromArgs(logGroupInfo, parameters)
+            const initialStreamData: CloudWatchLogsData = {
+                data: [],
+                parameters: parameters,
+                logGroupInfo: logGroupInfo,
+                retrieveLogsFunction: getLogEventsFromUriComponents,
+                busy: false,
+            }
+            await this.registry.registerLog(streamUri, initialStreamData)
+            // Set the document language
+            const doc = await vscode.workspace.openTextDocument(streamUri)
+            vscode.languages.setTextDocumentLanguage(doc, 'log')
```

Comment:
This looks to be duplicated in several places. We'll want to factor this out.

Reply:
Yea, I tried to factor out the parts that were easy to. A lot of this code is in `prepareDocument`, but theres two key differences that don't allow me to reuse that. First, I don't want to show the document here, just load it. And secondly, I don't need to worry about cancellation here (can't cancel a jumpto action).